doc/kubernetes/health-check.md

```diff
@@ -0,0 +1,27 @@
+# Why need health-check
+- Current status
+    - We can only check whether the k8s scheduled container is running or not, couldn't exactly know the real supervisor service inside container is healthy or not
+- Problems should be resolved
+    - Health-check should check whether the real supervisor service is healthy or not
+    - If not healthy, there better be different error codes to help debug
+
+# How do we Implement the health-check 
+- Leverage k8s probe tool
+    - K8s has three kinds of probe, [liveness, readiness and startup probes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes) which can help us probe state inside container. we can use startup probe for our scenario.
+    - Startup probe has four probe types, http|TCP|gRPC|command, command type should be good for our scenario.
```

Comment:
not only startup probe, each probe supports those probe types.

Reply:
Yes, I mean we plan use startup probe, so introduce the startup probe only. More details can be found in the link.